src/lib/utils/pcbManualEditEventHandler.ts

```diff
@@ -46,22 +46,22 @@ export const applyPcbEditEvents = (
     const handledEventIds = new Set<string>()
 
     // Add existing event IDs to the set
-    newState.pcb_placements.forEach((placement) => {
+    newManualEditState.pcb_placements.forEach((placement) => {
       if (placement._edit_event_id) {
         handledEventIds.add(placement._edit_event_id)
       }
     })
 
-    newState.edit_events.forEach((event) => {
+    newManualEditState.edit_events.forEach((event) => {
       if (event.edit_event_id) {
         handledEventIds.add(event.edit_event_id)
       }
     })
 
     // Process new edit events
-    for (const editEvent of editEvents) {
+    for (const editEvent of changedEditEvents) {
       if (
-        !editEvent.edit_event_id ||
+        (editEvent.in_progress && !editEvent.edit_event_id) ||
```

Comment:
@seveibar this condition is checking for the events that are complete, and only then proceed. Thus reduces the re-renders.

Reply:
nice